src/level-editor/editorState.ts

```diff
@@ -1,33 +1,17 @@
-import { createReducer, arraylike, settable } from '@dwalter/create-reducer'
+import { createReducer, settable } from '@dwalter/create-reducer'
 import { Resolve } from '@dwalter/spider-hook'
 import { Dispatch } from '@dwalter/spider-hook/src/types'
-
-export type Location = [number, number]
-
-export interface Tile {
-  location: Location
-  color: 'white' | 'black'
-}
-
-const [reducer, actions] = createReducer(
-  'tiles',
-  [{ location: [3, 4], color: 'white' }],
-  arraylike<Tile>(),
-)
-
-export const tilesSelector = reducer
-
-export const addTile = actions.add
+import { Vec2, Tile, tilesSelector, tileActions } from '../entity'
 
 const [goatReducer, goatActions] = createReducer(
```

Comment:
I'm going to make an issue and assign it to you to make `Goat` into a proper entity like I did with `Tile` so you can see how all this crap works a bit better